### DISCUSSION TITLE
Why is Bun slower than Node when running a loop with an index larger than 1_000_000?

### DISCUSSION THREAD

**Discussion by dinhchien-zps** [oven-sh/bun#10945] [Q&A]
When I run a loop with an index smaller than 1_000_000, Bun is still faster than Node:
![image](https://github.com/oven-sh/bun/assets/80091601/e85c0f4e-be6f-4bf7-ba7f-af385f209d63)
But when I run a loop with an index larger than 1_000_000, Bun is much slower than Node:
![image](https://github.com/oven-sh/bun/assets/80091601/9e512ab4-2fdf-473c-9281-a31e46b17004)
Can someone explain to me why?

**Comment by truongan07** [comment]
![image](https://github.com/oven-sh/bun/assets/141234261/5212d17e-dd6d-4595-9908-1c6f58024803)

![image](https://github.com/oven-sh/bun/assets/141234261/bf266e73-ae34-4ef8-acf3-d822772d8d3d)

The gap between Bun & Node is trivial

**Reply by dinhchien-zps** [reply]
Increase to 1B, and you will see a big gap.

**Reply by truongan07** [reply]
You want to iterate 1B elements of array ?

**Reply by dinhchien-zps** [reply]
I just want to know if Bun is really faster than Node. When an array has 1B elements, Bun is 3x slower than Node. Why's that?

**Reply by truongan07** [reply]
- See : 
```js
const computing = items => {
    const list = [],
        fnA = item => {
            // do something with item and return an array. This function completes in 3s by get data from database

            return [];
        },
        fnB = (item, resultA) => {
            // do something with params and return an object. This function completes in 200ms

            return {};
        };

    for (const item of items) {
        const resultA = fnA(item), resultB = fnB(item, resultA);

        list.push(resultB);
    }

    return list;
};

// items is an array contains 1B elements
computing(items);
```

- From script in the fact. Do you think speeding of loop is important ?

**Comment by alexeyten** [comment]
You are comparing optimization of empty loop in V8 vs JavaScriptCore.

Any numbers you could get from this comparison will have nothing to do with real world scenarios.

**Comment by e3dio** [comment]
I get `210 ms` for Bun v1.1.8 and `312 ms` for Node v22 for 1 Billion iteration loop, which means Bun is 33% faster than Node in this case. What system are you on, looks like Windows ? Try Linux much better OS

**Reply by g-cqd** [reply]
As much as we get the point of what you're saying. Giving an advice to switch your OS is definitely not a reasonable answer/advice.

That said, maybe he should try macOS instead. /j

Anyway, I'm interested in knowing the outcome of this experiment and the conclusions.

**Reply by e3dio** [reply]
If you are interested in performance or just sane computing in general advising to switch to Linux is definitely a reasonable suggestion
